mlflow/utils/docstring_utils.py

```diff
@@ -0,0 +1,90 @@
+import textwrap
+import re
+
+
+LOG_MODEL_PARAM_DOCS = {
+    "pip_requirements": """
+Either an iterable of pip requirement strings
+(e.g. ``["scikit-learn", "-r requirements.txt"]``) or the string path to a pip requirements
```

Comment:
Nit: It may be a little confusing to use `scikit-learn` as the example for all MLflow Model flavors. If it's not too hard, it might be nice to sub in different library names here. If that's too difficult, that's okay!

Reply:
What about `pandas`?

Reply:
For sklearn, the param doc should look like:

```
:param pip_requirements: ... [scikit-learn, -r requirements.txt]
```

but, for xgboost, the param doc should look like:

```
:param pip_requirements: ... [xgboost, -r requirements.txt]
```

Is this what you mean by `sub in different library names here`?

Reply:
@harupy `pandas` or `numpy` is better for `extra_pip_requirements`. For XGBoost, it would be nice if `pip_requirements` included the library name. If this is too time consuming, let's just use `pandas` or `numpy` for all of them.

Reply:
I think we can introduce a class like this:

```python
class ParamDocs(dict):
    def format(self, **replacements):
        new_param_docs = {}
        for param_name, param_doc in self.items():
            for name, replacement in replacements.items():
                placeholder = "{{ " + name + " }}"
                new_doc = param_doc.replace(placeholder, replacement)
                new_param_docs[param_name] = new_doc

        return ParamDocs(new_param_docs)
```

which behaves like `dict` and is capable of replacing placeholders.